layouts/partials/posts_navigation.html

```diff
@@ -1,15 +1,23 @@
 {{ $skipPageNav := .Scratch.Get "skipPageNav" }}
 {{ if not (or .Params.skip_pagenav $skipPageNav) }}
-    {{ if or .NextInSection .PrevInSection }}
+    {{$prev := .PrevInSection}}
+    {{$next := .NextInSection}}
+    {{ with .Params.prevInSection }}
+        {{$prev = .}}
```

Comment:
`prevInSection` [returns a Page object](https://gohugo.io/methods/page/previnsection/).  
The front matter contains a text string pointing to a page.

You need to get the object for that page with a `{{- with $.Site.GetPage . -}}` block, and inside _that_ you can assign `.` to `$prev`.

Look at the Hextra example again if you need to.